xs2a-impl/src/test/java/de/adorsys/psd2/xs2a/service/authorization/ais/DecoupledAisAuthorizationServiceTest.java

```diff
@@ -0,0 +1,211 @@
+import java.util.Optional;
+
+import static de.adorsys.psd2.xs2a.config.factory.AisScaStageAuthorisationFactory.SEPARATOR;
+import static de.adorsys.psd2.xs2a.config.factory.AisScaStageAuthorisationFactory.SERVICE_PREFIX;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.when;
+
+
+@RunWith(MockitoJUnitRunner.class)
+public class DecoupledAisAuthorizationServiceTest {
+    private static final String CONSENT_ID = "f2c43cad-6811-4cb6-bfce-31050095ed5d";
+    private static final String AUTHORISATION_ID = "a01562ea-19ff-4b5a-8188-c45d85bfa20a";
+    private static final PsuIdData PSU_DATA = new PsuIdData("psuId", "psuIdType", "psuCorporateId", "psuCorporateIdType");
+    private static final AccountConsentAuthorization ACCOUNT_CONSENT_AUTHORIZATION = buildAccountConsentAuthorization();
+    private static final ScaStatus SCA_STATUS = ScaStatus.STARTED;
+    private static final ScaApproach SCA_APPROACH = ScaApproach.DECOUPLED;
+    private static final List<String> STRING_LIST = Collections.singletonList(AUTHORISATION_ID);
+    private static final AccountConsent ACCOUNT_CONSENT = createConsent(CONSENT_ID);
+    private static final Xs2aAuthorisationSubResources AUTHORISATION_SUB_RESOURCES = new Xs2aAuthorisationSubResources(STRING_LIST);
+    private static final CreateConsentAuthorizationResponse CREATE_CONSENT_AUTHORIZATION_RESPONSE = buildCreateConsentAuthorizationResponse();
+    private static final UpdateConsentPsuDataReq UPDATE_CONSENT_PSU_DATA_REQ = new UpdateConsentPsuDataReq();
+    private static final UpdateConsentPsuDataResponse UPDATE_CONSENT_PSU_DATA_RESPONSE = new UpdateConsentPsuDataResponse();
+    private static final AisScaStage<UpdateConsentPsuDataReq, UpdateConsentPsuDataResponse> AIS_SCA_STAGE = null;
+
+    @InjectMocks
+    private DecoupledAisAuthorizationService decoupledAisAuthorizationService;
+
+    @Mock
+    private Xs2aAisConsentService aisConsentService;
+    @Mock
+    private Xs2aAisConsentMapper aisConsentMapper;
+    @Mock
+    private AisScaStageAuthorisationFactory scaStageAuthorisationFactory;
+    @Mock
+    private AisScaAuthenticatedStage aisScaAuthenticatedStage;
+
+    @Test
+    public void createConsentAuthorization_success() {
+        //Given
+        when(aisConsentService.getAccountConsentById(CONSENT_ID))
+            .thenReturn(ACCOUNT_CONSENT);
+        when(aisConsentService.createAisConsentAuthorization(CONSENT_ID, SCA_STATUS, PSU_DATA))
+            .thenReturn(Optional.of(AUTHORISATION_ID));
+
+        //When
+        Optional<CreateConsentAuthorizationResponse> actualResponse = decoupledAisAuthorizationService.createConsentAuthorization(PSU_DATA, CONSENT_ID);
+
+        //Then
+        assertThat(actualResponse.get().getAuthorizationId()).isEqualTo(AUTHORISATION_ID);
+        assertThat(actualResponse.get()).isEqualTo(CREATE_CONSENT_AUTHORIZATION_RESPONSE);
+    }
+
+    @Test
+    public void createConsentAuthorization_failed() {
```

Comment:
I would name this method like 
`createConsentAuthorization_wrongId` 
It will show what wrong in test